src/Backend/Modules/MediaLibrary/Domain/MediaFolder/MediaFolder.php

```diff
@@ -0,0 +1,331 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Domain\MediaFolder;
+
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Common\Collections\ArrayCollection;
+
+/**
+ * MediaFolder
+ *
+ * @ORM\Entity(repositoryClass="Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolderRepository")
+ * @ORM\HasLifecycleCallbacks
+ */
+class MediaFolder
+{
+    /**
+     * @var string
+     *
+     * @ORM\Id()
+     * @ORM\GeneratedValue(strategy="AUTO")
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @var MediaFolder
+     *
+     * @ORM\ManyToOne(
+     *      targetEntity="Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolder",
+     *      inversedBy="children",
+     *      cascade="persist"
+     * )
+     * @ORM\JoinColumn(
+     *      name="parentMediaFolderId",
+     *      referencedColumnName="id",
+     *      onDelete="cascade"
+     * )
+     */
+    protected $parent;
+
+    /**
+     * @var integer
+     *
+     * @ORM\Column(type="integer")
+     */
+    protected $userId;
+
+    /**
+     * @var string
+     *
+     * @ORM\Column(type="string")
+     */
+    protected $name;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $createdOn;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $editedOn;
+
+    /**
+     * @var ArrayCollection
+     *
+     * @ORM\OneToMany(
+     *     targetEntity="Backend\Modules\MediaLibrary\Domain\MediaItem\MediaItem",
+     *     mappedBy="folder",
+     *     cascade={"persist","merge"},
+     *     orphanRemoval=true
+     * )
+     */
+    protected $items;
+
+    /**
+     * @var ArrayCollection
+     *
+     * @ORM\OneToMany(
+     *     targetEntity="Backend\Modules\MediaLibrary\Domain\MediaFolder\MediaFolder",
+     *     mappedBy="parent",
+     *     cascade={"persist","merge"},
+     *     orphanRemoval=true
+     * )
+     */
+    protected $children;
+
+    /**
+     * Construct
+     *
+     * @param string $name The name of this folder.
+     * @param MediaFolder|null $parent The parent of this folder, can be NULL.
+     * @param int $userId The id of the user who created this MediaFolder.
+     */
+    protected function __construct(
+        string $name,
+        MediaFolder $parent = null,
+        int $userId
+    ) {
+        $this->name = $name;
+        if ($parent !== null) {
```

Comment:
obsolete check :)
if the parent is null you won't set it and it will be null, so you might as well set it :)